variants/github_actions_ci/workflows/deploy_to_ec2.yml

```diff
@@ -0,0 +1,78 @@
+name: deploy_to_ec2
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+      assume_role_arn:
+        type: string
+        required: true
+      aws_region:
+        type: string
+        required: true
+      timeout_minutes:
+        type: number
+        required: false
+        default: 15
+    secrets:
+      ssh_private_key:
+        required: true
+      slack_webhook:
+        required: true
```

Comment:
If we're going to make this required then I think we need to make sure we're got a common webhook in our secrets manager + documented somewhere since adding webhooks typically require workspace owner permissions on Slack sometimes.

Reply:
Yeah I'm unsure about the Slack stuff. It's nice to have in there because creating it from scratch is a bit of faff but it's not something every project will want by default.

I could leave the slack stuff in these files but commented out or just remove it. Currently I'm leaning towards here but commented.

Reply:
or we can make it optional and change the conditionals conditions to check if `slack_webhook` is set (should just be `if: ${{ secrets.slack_webhook && success() }}`).

I'd also be in favor of having one slack step since we can pass `job.status` for the color + use that in the message too; it'll be a little stiffer because it'll need to be rephrased to something like `${{ format('{0} job result: {1}', github.job, job.status) }}` but it would mean less complexity in the workflows.

tbh I think we should probably enable them on all projects: while they add noise, it makes it a bit easier for non-devs to know when a production deployment has finished.

Also I just realised we should be doing the `SLACK_WEBHOOK` as an organisation level secret

Reply:
That makes sense. I'll refactor.